### PR TITLE
Create v0.33.0 RC PR

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane_lightning/core/src/simulators/lightning_gpu/algorithms/AdjointJacobianGPUMPI.hpp [hunk 1]
```diff
@@ -235,6 +238,8 @@ class AdjointJacobianMPI final
         if (!jd.hasTrainableParams()) {
             return;
         }
+        MPIManager mpi_manager_(jd.getMPIManager());
```
Comment (mlxd) [inline pennylane_lightning/core/src/simulators/lightning_gpu/algorithms/AdjointJacobianGPUMPI.hpp:241]: Is there a need for all of the blocking calls here? Or is this the required fix from earlier? What PR added these lines?

File: requirements.txt [hunk 2]
```diff
@@ -5,3 +5,4 @@ pybind11
 pytest
 pytest-cov
 pytest-mock
+custatevec-cu11
```
Comment (mlxd) [inline requirements.txt:8]: Can we remove this please, since it is very arch specific.

File: requirements-dev.txt [hunk 3]
```diff
@@ -4,9 +4,11 @@ ninja
...
 pytest
+pytest-benchmark
 pytest-cov
 pytest-mock
 pre-commit>=2.19.0
 black==23.7.0
 clang-format==14
-pylint
\ No newline at end of file
+custatevec-cu11
```
Comment (mlxd) [inline requirements-dev.txt:13]: Fine to keep here, as we use it for docs/dev building

**Review by mlxd — DISMISSED** [review]
Quick review

File: .github/workflows/tests_linux_x86_mpi_gpu.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@ on:
   push:
     branches:
       - main
-  pull_request:
+  #pull_request:
```
Comment (multiphaseCFD) [inline .github/workflows/tests_linux_x86_mpi_gpu.yml:17]: Do we want to switch it on?

File: pennylane_lightning/core/src/simulators/lightning_gpu/algorithms/AdjointJacobianGPUMPI.hpp (hunk 1 above)
Comment (multiphaseCFD) [reply pennylane_lightning/core/src/simulators/lightning_gpu/algorithms/AdjointJacobianGPUMPI.hpp:241]: Not really. Only few lines needed. Will update it later after retro.

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -28,10 +31,10 @@
 
 ### Breaking changes
 
-* Add `tests_gpu.yml` workflow to test the Lightning-Kokkos backend with CUDA-12. 
+* Add `tests_gpu.yml` workflow to test the Lightning-Kokkos backend with CUDA-12.
   [(#494)](https://github.com/PennyLaneAI/pennylane-lightning/pull/494)
 
-* Implement `LM::GeneratorDoubleExcitation`, `LM::GeneratorDoubleExcitationMinus`, `LM::GeneratorDoubleExcitationPlus` kernels. L-Qubit default kernels are now strictly from the `LM` implementation, which requires less memory and is faster for large state vectors.  
+* Implement `LM::GeneratorDoubleExcitation`, `LM::GeneratorDoubleExcitationMinus`, `LM::GeneratorDoubleExcitationPlus` kernels. L-Qubit default kernels are now strictly from the `LM` implementation, which requires less memory and is faster for large state vectors.
```
Comment (AmintorDusko) [inline .github/CHANGELOG.md:37]:
```suggestion
* Implement `LM::GeneratorDoubleExcitation`, `LM::GeneratorDoubleExcitationMinus`, `LM::GeneratorDoubleExcitationPlus` kernels. Lightning-Qubit default kernels are now strictly from the `LM` implementation, which requires less memory and is faster for large state vectors.
```

File: .github/CHANGELOG.md [hunk 6]
```diff
@@ -92,7 +95,7 @@
 * Switch most L-Qubit default kernels to `LM`. Add `LM::multiQubitOp` tests, failing when targeting out-of-order wires clustered close to `num_qubits-1`. Fix the `LM::multiQubitOp` kernel implementation by introducing a generic `revWireParity` routine and replacing the `bitswap`-based implementation. Mimic the changes fixing the corresponding `multiQubitOp` and `expval` functors in L-Kokkos.
```
Comment (AmintorDusko) [inline .github/CHANGELOG.md:95]:
```suggestion
* Switch most Lightning-Qubit default kernels to `LM`. Add `LM::multiQubitOp` tests, failing when targeting out-of-order wires clustered close to `num_qubits-1`. Fix the `LM::multiQubitOp` kernel implementation by introducing a generic `revWireParity` routine and replacing the `bitswap`-based implementation. Mimic the changes fixing the corresponding `multiQubitOp` and `expval` functors in Lightning-Kokkos.
```

File: README.rst [hunk 7]
```diff
@@ -137,184 +189,214 @@ One can also build the plugin using CMake:
...
+On linux systems, ``lightning.kokkos`` with the OpenMP backend can be installed by providing the optional ``[kokkos]`` tag:
 
-    docker run -v `pwd`:/io -it quay.io/pypa/manylinux2014_x86_64 bash
+.. code-block:: console
 
-Next, within the container, we install the ROCm software stack:
+    $ pip install pennylane-lightning[kokkos]
 
-.. code-block:: console
+Install L-Kokkos from source
```
Comment (AmintorDusko) [inline README.rst:316]:
```suggestion
Install Lightning-Kokkos from source
```

**Review by AmintorDusko — COMMENTED** [review]
Just a few comments.

File: README.rst (hunk 7 above)
Comment (mlxd) [reply README.rst:316]: Before merging this commit, can we check that it doesn't impact the RST/Sphinx references during the docs-build? `rstcheck` and/or `twine check` against a valid source distribution wheel will tell us.

Comment (vincentmr) [reply README.rst]: Checked, I saw no issue related to those titles with either tool.

**Comment by mlxd** [comment]
Confirmed I am happy with the additions for the release.

**Comment by vincentmr** [comment]
Ready for merge!

**Comment by multiphaseCFD** [comment]
> Ready for merge!

Hey @vincentmr , Let me create a new branch and make a new PR as suggested by @mlxd. Will tag you soon

**Comment by multiphaseCFD** [comment]
Just want to confirm that we want to put [4d9a5d5](https://github.com/PennyLaneAI/pennylane-lightning/pull/530/commits/4d9a5d53c8169b9902c3283a4b72468a3a903f4c) to v0.33, while others to v0.34.0-dev.

**Comment by vincentmr** [comment]
If you do `git log v0.33.0` you'll see [4d9a5d5](https://github.com/PennyLaneAI/pennylane-lightning/pull/530/commits/4d9a5d53c8169b9902c3283a4b72468a3a903f4c) is already in v0.33. I think the whole history of `v0.33.0_rc` can then go into v0.34.

**Pull request closed without merging**